### ISSUE TITLE
List Folder Contents Permissions issue

### ISSUE THREAD

**Issue by nickalleyne** [raandree/NTFSSecurity#13]
Hi there,

I have just begun using this module, and so far it has been working great, so thanks for putting it together.

The only issue I have run into so far, is that I cannot seem to properly set "List folder contents" as you would via the gui.

For example, if I choose a folder in the gui, and add a new user with only "List folder contents", then it applies via the gui and all seems to work as expected.  If when running Get-NTFSAccess I see it shows "ReadAndExecute" + "Synchronize" (below output sanitized).

PS C:\Test> Get-NTFSAccess -Path .\User1\ -Account fileshare-list | fl

Name               : User1
FullName           : C:\Test\User1
InheritanceEnabled : True
InheritedFrom      :
AccessControlType  : Allow
AccessRights       : ReadAndExecute, Synchronize
Account            : Fileshare-List
InheritanceFlags   : ContainerInherit
IsInherited        : False
PropagationFlags   : None
AccountType        : group

But if I then go and re-add the same permissions (or delete and re-create), using ReadAndExecute, it gives ReadAndExecute (as expected).

I have tried adding just "ListDirectory" but then in the gui it shows "Special" which will confuse my helpdesk when they see it.  I have also tried a number of other combinations of attributes, but nothing seems to get just the "List folder contents" box to be checked.

In case it changes anything, I am running on Windows 10 1703 (or 1704, cannot remember which) Creators Update.  I have tried this both on local and network shares, and I get the same result.

Thanks again

**Comment by raandree** [comment]
The reason is that the enum that contains the access rights definitions has duplicate values:

``` PowerShell
#region function Get-Enum
function Get-Enum { 
    param (
        [type]$Type
    )

    [enum]::GetNames($Type) | 
    Select-Object -Property `
    @{ Name = 'Name'; Expression={ [string]$_ } },
    @{ Name = 'Value'; Expression={ [uint32](Invoke-Expression "[$($type.FullName)]'$_'") }},
    @{ Name = 'Binary'; Expression={[Convert]::ToString([uint32](Invoke-Expression "[$($type.FullName)]'$_'"), 2)}}
}
#endregion

Get-Enum -Type System.Security.AccessControl.FileSystemRights
```

I will look into this...

**Comment by raandree** [comment]
I guess the Explorer is cheating on us. I cannot really give just the right "ListDirectory", it always comes with "Read". I have removed everything from the access entry for some user except from "List folder content":
![image](https://cloud.githubusercontent.com/assets/11280760/25962594/9d4022d0-367e-11e7-9cf0-753bb027e4eb.png)

Get-NTFSAccess returns this:
``` PowerShell
PS D:\Test> (Get-NTFSAccess -Account install).AccessRights
ReadAndExecute, Synchronize
```

And this is true if you look up the advanced security settings in the explorer:
![image](https://cloud.githubusercontent.com/assets/11280760/25962649/d304bb06-367e-11e7-886e-fa2cd2c2c8f2.png)

And here it is simply not possible to give someone just "List folder content". It always comes with "Read" and "Read & Execute".
And ReadAndExecute (131241) is a compination of:

- ReadPermissions 131072
- ReadAttributes 128
- ExecuteFile or Traverse, both 32
- ReadExtendedAttributes 8
- ListDirectory or ReadData, both 1

So, bottom line is, to achive what the Windows Explorer does, you want to do this:

``` PowerShell
Add-NTFSAccess -Path D:\Test -Account Install -AccessRights ReadAndExecute
```

**Comment by nickalleyne** [comment]
Thanks so much for looking into that.  I can't believe I did not notice it when checking advanced and doing it manually via the GUI.

**Comment by raandree** [comment]
Great I could help. I am closing the issue then.

**Comment by rodmhgl** [comment]
My understanding is that List Folder Contents was just Read & Execute applied to Folders only.
